### ISSUE TITLE
x-vector generation

### ISSUE THREAD

**Issue by klh5** [alan-turing-institute/ARC-Speech-Synthesis#11]
- What exactly are x-vectors?
- What is the easiest/fastest way to generate them for a set of speech samples?

**Comment by klh5** [comment]
**x-vectors** are described in the original paper [here](https://www.danielpovey.com/files/2017_interspeech_embeddings.pdf). Briefly:

- A DNN is trained on a multi-class cross-entropy objective (in the original paper there are 4733 speakers, so that is the size of the softmax output layer)
- The inputs are Mel Frequency Cepstral Coefficients (MFCCs) derived from the speech samples
    - MFCCs are a set of features (usually 10-20) which describe the overall shape of the [spectral envelope](https://timbreandorchestration.org/writings/timbre-lingo/spectral-envelope). These features are notionally supposed to describe _timbre_, but as that is itself not a well defined concept, it's difficult to say exactly what they represent
    - MFCCs seem to be the reduced set of coefficients from a Discrete Cosine Transform of the mel spectrogram - see [this description](https://medium.com/@derutycsl/intuitive-understanding-of-mfccs-836d36a1f779)
- The MFCCs are fed through the DNN to extract segment-level statistics (mean and SD) from speech frames, which are then fed into some additional layers to obtain embeddings
- The embeddings are reduced using Linear Discriminant Analysis

Prior to the paper above in 2017, **i-vectors** were the go-to method for extracting speaker features using Gaussian Mixture Models.

**Comment by klh5** [comment]
The x-vectors for [CMU Arctic](https://huggingface.co/datasets/Matthijs/cmu-arctic-xvectors) used in the SpeechT5 tutorial (see #7) are generated using [this model](https://huggingface.co/speechbrain/spkrec-ecapa-voxceleb) from [SpeechBrain](https://huggingface.co/speechbrain). I don't know yet whether this is a standard approach.

**Comment by klh5** [comment]
Closing as I think the pipeline we have is sufficient for now.
